<?xml version="1.0"?>
<launch>
    <!-- ============================ SETTINGS ============================= -->

    <!-- Node frequency: the frequency [Hz] which rate.sleep() function will
    attempt to maintain in the main_loop. -->
    <arg 
    name="node_frequency" 
    default="100"/>

    <arg 
    name="robot_name" 
    default="my_gen3"/>

    <!-- If any of dependency or optional subfiles should not be launched (or
    are launched separately) set corresponding argument to "false". -->
    <arg 
    name="launch_teleoperation" 
    default="true"/>
    <arg 
    name="launch_controller_feedback" 
    default="true"/>

    <!-- ======================== END OF SETTINGS ========================== -->

    <!-- Teleoperation module (dependency): -->
    <group if="$(arg launch_teleoperation)">
        <include 
        file="$(find kinova_positional_control)/launch/teleoperation.launch">
            <arg 
            name="robot_name" 
            value="$(arg robot_name)"/>
        </include>
    </group>

    <!-- Oculus controller feedback (dependency): -->
    <group if="$(arg launch_controller_feedback)">
        <include 
        file="$(find oculus_ros)/launch/controller_feedback.launch">
            <arg 
            name="controller_side" 
            value="right"/>
            <arg 
            name="enable_joystick_deadzones" 
            value="true"/>
        </include>

        <include 
        file="$(find oculus_ros)/launch/controller_feedback.launch">
            <arg 
            name="controller_side" 
            value="left"/>
            <arg 
            name="enable_joystick_deadzones" 
            value="true"/>
        </include>
    </group>

    <!-- Oculus joystick mapping: -->
    <group ns="/$(arg robot_name)">
        <node 
        pkg="kinova_positional_control" 
        type="oculus_joystick_mapping.py" 
        name="oculus_joystick_mapping" 
        output="screen">
            <param 
            name="node_frequency" 
            value="$(arg node_frequency)"/>
            <param 
            name="robot_name" 
            value="$(arg robot_name)"/>
        </node>
    </group> 
</launch>